src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -995,41 +1005,22 @@ async Task ConnectAsync(NpgsqlTimeout timeout, CancellationToken cancellationTok
             }
         }
 
-        Task<IPAddress[]> GetHostAddressesAsync(NpgsqlTimeout timeout, CancellationToken cancellationToken)
-        {
-            // .NET 6.0 added cancellation support to GetHostAddressesAsync, which allows us to implement real
-            // cancellation and timeout. On older TFMs, we fake-cancel the operation, i.e. stop waiting
-            // and raise the exception, but the actual connection task is left running.
-
-#if NET6_0_OR_GREATER
-            var task = TaskExtensions.ExecuteWithTimeout(
-                ct => Dns.GetHostAddressesAsync(Host, ct),
-                timeout, cancellationToken);
-#else
-                var task = Dns.GetHostAddressesAsync(Host);
-#endif
-
-            // As the cancellation support of GetHostAddressesAsync is not guaranteed on all platforms
-            // we apply the fake-cancel mechanism in all cases.
-            return task.WithCancellationAndTimeout(timeout, cancellationToken);
-        }
-
-        static Task OpenSocketConnectionAsync(Socket socket, EndPoint endpoint, NpgsqlTimeout perIpTimeout, CancellationToken cancellationToken)
-        {
-            // .NET 5.0 added cancellation support to ConnectAsync, which allows us to implement real
-            // cancellation and timeout. On older TFMs, we fake-cancel the operation, i.e. stop waiting
-            // and raise the exception, but the actual connection task is left running.
-
+            static Task OpenSocketConnectionAsync(Socket socket, EndPoint endpoint, NpgsqlTimeout perIpTimeout, CancellationToken cancellationToken)
+            {
+                // .NET 5.0 added cancellation support to ConnectAsync, which allows us to implement real
+                // cancellation and timeout. On older TFMs, we fake-cancel the operation, i.e. stop waiting
+                // and raise the exception, but the actual connection task is left running.
+                Task ConnectAsync(CancellationToken ct)
```

Comment:
A local function here seems a bit heavy - it seems simpler to just do something like:

```c#
#if NET5_0_OR_GREATER
                var connectTask = socket.ConnectAsync(endpoint, cancellationToken).AsTask();
#else
                var connectTask = socket.ConnectAsync(endpoint);
#endif
```

Reply:
The reason for a local function is the same as explained [here](https://github.com/npgsql/npgsql/pull/4173/files#r776901310)

Reply:
Makes sense, thanks.